components/calculation/PopulationComparisonTable.tsx

```diff
@@ -47,45 +47,193 @@ export default function PopulationComparisonTable({ patientId }: PopulationCompa
-      const key = result?.criteriaExpression;
-      if (key) {
-        bothPopulations[key as string] = {
-          desired: desiredPopulations[key as string],
-          actual: result?.result === true ? 1 : 0
-        };
-      }
+  function constructPatientValues(group: DetailedPopulationGroupResult | undefined) {
+    const desiredPopulations = constructPatientDesiredValuesArray(getMeasurePopulationsForSelection(measure));
+    const patientValues: ResultValues = {
+      resource: 'Patient',
+      desired: {},
+      actual: {}
+    };
+    // TODO: assumes one group, may need updating if there's more than one group
+    let scoreType = measure.group?.[0].extension?.find(
+      e => e.url === 'http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-scoring'
+    )?.valueCodeableConcept?.coding?.[0].code;
+    if (!scoreType) {
+      // Score type might also be in the measure scoring field
+      scoreType = measure.scoring?.coding?.[0].code;
+    }
+    if (scoreType === 'proportion') {
+      // generate boolean population scores for proportion measures
+      group?.populationResults?.forEach(result => {
+        const key = keyForResult(result, group);
+
+        patientValues['actual'][key] = result?.result === true ? 1 : 0;
+        patientValues['desired'][key] = desiredPopulations[key];
+      });
+    } else if (scoreType === 'ratio') {
+      // generate number of episodes for each population for ratio measures
+      group?.episodeResults?.forEach(ep_result => {
+        ep_result.populationResults.forEach(result => {
+          const key = keyForResult(result, group);
+
+          if (result.populationType === 'measure-observation') {
+            patientValues['actual'][key] = undefined;
+            patientValues['desired'][key] = undefined;
+          } else {
+            // TODO: placeholder 0 value until we can set a desired total number of episodes per population
+            patientValues['desired'][key] = 0;
+            if (!patientValues['actual'][key]) patientValues['actual'][key] = 0;
+            if (result.result) {
+              patientValues['actual'][key] = (patientValues['actual'][key] as number) + 1;
+            }
+          }
+        });
+      });
+    } else {
+      throw new Error('Could not find proportion or ratio score type.');
+    }
+    return patientValues;
+  }
+
+  /**
+   * Creates an array of ResultValues objects for each of the Episode results. Specifies resource (which episode),
+   * desired population values, and actual population values. Desired and actual are key/value pairs,
+   * where the population label (criteria) is the key and the value is 0/1 or the observation value if the population
+   * is a measure observation.
+   */
+  function constructEpisodeValues(group: DetailedPopulationGroupResult | undefined) {
+    const results = group?.episodeResults?.map(er => {
+      const episode = currentPatients[patientId].resources.find(r => r.id === er.episodeId)?.resource;
```

Comment:
```suggestion
      const episode = currentPatients[patientId].resources.find(r => r.resource?.id === er.episodeId)?.resource;
```

When I upload the measure bundle from the fqm-execution integration test file and import a patient from that file, the episode result labels are undefined: undefined: 

![image](https://user-images.githubusercontent.com/30158156/236935930-9182597e-210f-4f75-9fb6-08afc715b5dc.png)


I did some quick digging and I think it's because id isn't accessed on the resource but actually r.resource.id.

Reply:
![image](https://user-images.githubusercontent.com/30158156/236936276-6568fb6b-039c-49d0-8944-dbb505ca0c02.png)

That seemed to fix the last three

Reply:
I think there should only be three though...

Reply:
Is this resolved at this point? Not sure if you were still seeing issues after the [Fix resource name issue (finish rebase fix)](https://github.com/projecttacoma/fqm-testify/pull/67/commits/f481be6e4ce466893b2f7fc0394cca01f3d91b28) update?